emulators/qemu/Portfile

```diff
@@ -71,6 +71,12 @@ patchfiles-append       patch-qemu-include-qemu-madvise_h.diff
 # MAP_JIT workaround
 patchfiles-append       patch-qemu-tcg-region.diff
 
+if {[string match *gcc* ${configure.compiler}]} {
+    # https://github.com/macports/macports-ports/pull/14807#issuecomment-1184586924
+    # FIXME: verify if the fix is doing what it is supposed to.
+    patchfiles-append   patch-VLA.diff
+}
+
 # "ERROR: You need at least GCC v7.5 or Clang v6.0 (or XCode Clang v10.0)"
 compiler.blacklist      {clang < 1000} {macports-clang-[3-5].*} {macports-gcc-[56]} {*gcc-[34].*}
 if {${os.platform} eq "darwin" && ${os.major} < 11 && [string match *clang* ${configure.compiler}]} {
```

Comment:
```suggestion
# "ERROR: You need at least GCC v7.5 or Clang v6.0 (or XCode Clang v10.0)"
compiler.blacklist      {clang < 1000} {macports-clang-[3-9].*} {macports-gcc-[56]} {*gcc-[34].*}
if {${os.platform} eq "darwin" && ${os.major} < 11 && [string match *clang* ${configure.compiler}]} {
```

I assume that MacPort's clang up to 9.0 should be blacklisted.

Reply:
> I assume that MacPort's clang up to 9.0 should be blacklisted.

No, not necessarily. Projects often base their requirements on Xcode Clang, which is often behind - sometimes way behind - the patchsets which we apply to our Clang versions.

So I'd actually be very surprised if MacPorts Clang 9 doesn't work...

Reply:
@mascguy it doesn't and meson fails :(

Reply:
or, maybe it was clang-7? Let me check.

Reply:
Yes, it was clang-7:

```
Executing:  cd "/opt/local/var/macports/build/_Volumes_SharedFolders_Home_src_macports-ports_emulators_qemu/qemu/work/build" && /opt/local/var/macports/build/_Volumes_SharedFolders_Home_src_macports-ports_emulators_qemu/qemu/work/qemu-8.0.4/configure --prefix=/opt/local --cpu=i386 --cc=/opt/local/bin/clang-mp-7.0 --objcc=/opt/local/bin/clang-mp-7.0 --host-cc=/opt/local/bin/clang-mp-7.0 --sphinx-build=/opt/local/bin/sphinx-build-3.11 --meson=/opt/local/bin/meson --iasl=/usr/bin/false --disable-attr --disable-brlapi --disable-cap-ng --disable-capstone --disable-cocoa --disable-curl --enable-curses --disable-dbus-display --disable-gcrypt --disable-glusterfs --disable-gtk --disable-jack --disable-libiscsi --disable-libssh --enable-libusb --disable-linux-aio --disable-nettle --disable-numa --disable-opengl --enable-png --disable-rbd --disable-rdma --disable-sdl --disable-seccomp --disable-smartcard --disable-spice --disable-spice-protocol --enable-usb-redir --disable-vde --enable-vnc --disable-xen --enable-bochs --enable-bzip2 --enable-cloop --enable-dmg --enable-gnutls --disable-hvf --enable-lzfse --enable-lzo --enable-parallels --enable-qcow1 --enable-qed --enable-snappy --enable-slirp --enable-tools --enable-vdi --disable-virtfs --enable-vvfat --enable-zstd --smbd=/opt/local/sbin/smbd --extra-cflags=-DNCURSES_WIDECHAR=1 --enable-vnc-jpeg --enable-vnc-sasl --target-list=,i386-softmmu,x86_64-softmmu,arm-softmmu,aarch64-softmmu --extra-ldflags="-Wl,-headerpad_max_install_names -L/opt/local/lib -lMacportsLegacySupport" 

ERROR: You need at least GCC v7.4 or Clang v10.0 (or XCode Clang v12.0)
```